test/jdk/java/awt/ScrollPane/ScrollPaneRemoveAdd.java

```diff
@@ -102,14 +103,66 @@ public void start() throws Exception {
             robot.delay(50);
             robot.waitForIdle();
             if (!actionSema.getState()) {
-                throw new RuntimeException("ScrollPane doesn't handle correctly add after remove");
+                throw new RuntimeException("ScrollPane doesn't handle " +
+                        "correctly add after remove");
             }
         } finally {
-            EventQueue.invokeLater(() -> {
+            EventQueue.invokeAndWait(() -> {
                 if (frame != null) {
                     frame.dispose();
                 }
             });
         }
     }
+
+    private static class Semaphore {
+        volatile boolean state = false;
+        Object lock = new Object();
```

Comment:
Since it's a lock object, it should be `final`.

Reply:
Updated.